action-processor/create-action.js

```diff
@@ -412,15 +412,15 @@ async function createSecureMessage (featureName, action){
     var sendData = {}
     sendData.Subject = action.data.Subject
     sendData.Message = action.data.Message
+    sendData.Attachments = action.data.Attachments
     sendData.FromId = response.Id
 
-    sendData.Recipients = []
+    sendData.RecipientIds = []
     for (const userLabel of action.data.Recipients) {
         let savedAction = await localStorage.getItem(featureName, userLabel)
-        let response = savedAction.response
-        let userEmail = response.Email
-        log.info(`${SERVICE_NAME}::createSecureMessage::Recipient ${userEmail}`)
-        sendData.Recipients.push(userEmail)
+        let userId = savedAction && savedAction.response && savedAction.response.Id
```

Comment:
`userId` will return a boolean rather than the actual value (it's testing if all those things have a value or not)

Reply:
it actually works
https://stackoverflow.com/a/32397039/5046575
or just try it in the browser console
document && document.title

Reply:
but if you suggest a refactoring I will use it

Reply:
OK, refactored in more readable syntax

Reply:
Maybe `if(savedAction && savedAction.response && savedAction. response.Id)` - in case `savedAction.response.Id` resolves to false?